### ISSUE TITLE
lua_minlstring type convert error

### ISSUE THREAD

**Issue by tinglei8** [kext/lua-nacl#1]
Hi, there is a horrible type convert error in the code: (took me hours to find out... so please fix this for other people's good)

``` c
static u8* lua_minlstring(lua_State *L, int index, int len, u64 *l, i8 *error)
 {
+  u8* res;
+  size_t slen;
   if (!lua_isstring(L, index) || lua_rawlen(L, index) < len)
   {
     lua_pushfstring(L, error, len);
     lua_error(L);
   }
-  return (u8*)lua_tolstring(L, index, (size_t*)l);
+  res = (u8*) lua_tolstring(L, index, &slen);
+  *l = (u64) slen;
+  return res;
 }
```

**Comment by tinglei8** [comment]
Sorry the format was ruined, the catch line is 

``` c
return (u8)lua_tolstring(L, index, (size_t *)l);
```

Shouldn't convert a `u64 *` to `size_t *`

**Comment by kext** [comment]
You are indeed correct. Thank you!

I’m pretty amazed that somebody is actually using this code. :D

Fixed by 08144b9155.
